### DISCUSSION TITLE
i18n - translation

### DISCUSSION THREAD

**Discussion by drarko** [renovatebot/renovate#14840] [Request Help]
Hi,

Does Renovate support any kind of localization? 

I need to translate messages to Spanish if a want to use it at my company, bc corporate policies

**Comment by rarkins** [comment]
Unfortunately not at this point. Which aspects would require translation - just the public facing parts (issues, PRs, etc) - or also all the logs too?

**Reply by drarko** [reply]
Logs aren't required to be translated, just public facing parts which have message or interactions with users

**Reply by rarkins** [reply]
Most user-facing strings are configurable these days, so potentially in future it could be done using presets. In fact if we went that way we'd probably want to isolate all such strings into an internal "english" preset so that it's easier to compare language coverage.

**Reply by drarko** [reply]
Thanks for clarification.

In meantime I will try to got a waiver from our Corporate counterpart to use Renovate, we have more than 1700 repositories and dependencies management is one of our principal and more harming pains, for DevOps and DevSecOps areas in particular (my direct reportees)

Also, we will start with translation, so in future we can collaborate with this localization. I will stay tuned to any advance in this way

Regards

**Comment by xingxing** [comment]
Hi, guys, what about using [gettext.js](https://github.com/guillaumepotier/gettext.js/), it seems like a good choose, because that  lib embraces fully the [gettext](https://www.gnu.org/software/gettext/) API and philosoph, that let people work on a same page. I'll give a PR for that ASAP.

**Reply by xingxing** [reply]
Oh, by the way, I am glad to support Chinese translation.

**Reply by rarkins** [reply]
Renovate has a lot of user-facing strings, e.g. in issues, PRs and check runs. Additionally commit messages, PR titles and log warnings also get publicly exposed.

**Reply by xingxing** [reply]
OK, that will be an epic work.

**Reply by xingxing** [reply]
I don't think log text should be translate actually. For a bot admin, he/she should totally understand the log messages.

**Reply by rarkins** [reply]
I mean that we intentionally add some warning log messages to PR bodies so that users are aware to check their logs for more details

**Reply by xingxing** [reply]
> I mean that we intentionally add some warning log messages to PR bodies so that users are aware to check their logs for more details

OK, I'll check that when I encounter that logic in coding. First things first is to improve friendly to users who want to adopt Renovate to their corporation, IMHO.
